Sources/Utility/Process.swift

```diff
@@ -165,6 +171,21 @@ public final class Process: ObjectIdentifierProtocol {
         self.redirectOutput = redirectOutput 
     }
 
+    /// Validate that the program provided exists on the search path and is executable.
+    func validateArgumentIsExecutable(_ program: String) throws {
+
+        // Check to see if the argument is already listed in the vaildated cache.
+        guard !Process.validatedExecutables.contains(program) else {
+            return
+        }
+        let envSearchPaths = Utility.getEnvSearchPaths(pathString: getenv("PATH"), currentWorkingDirectory: currentWorkingDirectory)
```

Comment:
The type of variable (class or property) depends on the use case, since we require that this method should be protected across all the instance of Process, it should be a class property.

Yes, by default, a serial queue is created and then you can dispatch work items on those queues. I haven't read this one specifically but tutorials here are generally good: https://www.raywenderlich.com/148513/grand-central-dispatch-tutorial-swift-3-part-1